lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -257,148 +266,262 @@ def common_problems(self, trans: ProvidesUserContext, id, **kwd):
+        are slightly different than dataset checks, so both methods are
+        available.
+
+        This API endpoint is unstable and tied heavily to Galaxy's JS client code,
+        this endpoint will change frequently.
         """
-        job = self.__get_job(trans, **kwd)
-        return summarize_job_metrics(trans, job)
+        job = self.service.get_job(trans, job_id=id, hda_ldda=hda_ldda)
+        return summarize_job_parameters(trans, job)
 
-    @require_admin
-    @expose_api
-    def destination_params(self, trans: ProvidesUserContext, **kwd):
+    @router.get(
+        "/api/datasets/{id}/parameters_display",
+        name="resolve_parameters_display",
+        summary="Resolve parameters as a list for nested display.",
+    )
+    def parameters_display_by_dataset(
+        self,
+        id: Annotated[DecodedDatabaseIdField, DatasetIdPathParam],
+        hda_ldda: Annotated[DatasetSourceType, HdaLddaQueryParam] = DatasetSourceType.hda,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> JobDisplayParametersSummary:
         """
-        * GET /api/jobs/{job_id}/destination_params
-            Return destination parameters for specified job.
-
-        :type   job_id: string
-        :param  job_id: Encoded job id
-
-        :rtype:     list
-        :returns:   list containing job destination parameters
+        # TODO is this still true?
+        Resolve parameters as a list for nested display. More client logic
+        here than is ideal but it is hard to reason about tool parameter
+        types on the client relative to the server. Job accessibility checks
+        are slightly different than dataset checks, so both methods are
+        available.
+
+        This API endpoint is unstable and tied heavily to Galaxy's JS client code,
+        this endpoint will change frequently.
         """
-        job = self.__get_job(trans, **kwd)
-        return summarize_destination_params(trans, job)
+        job = self.service.get_job(trans, dataset_id=id, hda_ldda=hda_ldda)
+        return summarize_job_parameters(trans, job)
 
-    @expose_api_anonymous
-    def parameters_display(self, trans: ProvidesUserContext, **kwd):
+    @router.get(
+        "/api/jobs/{id}/metrics",
+        name="get_metrics",
+        summary="Return job metrics for specified job.",
+    )
+    def metrics_by_job(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        hda_ldda: Annotated[DatasetSourceType, HdaLddaQueryParam] = DatasetSourceType.hda,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> List[Optional[JobMetric]]:
+        job = self.service.get_job(trans, job_id=id, hda_ldda=hda_ldda)
+        return [JobMetric(**metric) for metric in summarize_job_metrics(trans, job)]
+
+    @router.get(
+        "/api/datasets/{id}/metrics",
+        name="get_metrics",
+        summary="Return job metrics for specified job.",
+    )
+    def metrics_by_dataset(
```

Comment:
I think it makes sense to mark all the `*_by_dataset` as deprecated routes, they don't work for dataset collections and it's relatively indirect compared to getting information via the job.